smaug/python/node.py

```diff
@@ -0,0 +1,126 @@
+import sys
+import numpy as np
+
+from smaug.core import node_pb2
+from smaug.core import types_pb2
+from smaug.python import global_vars
+from smaug.python import datatypes
+
+class Node:
+  def __init__(self, name, op, params=None, inputs=None, outputs=None):
+    """Create a node.
+
+    Args:
+      name: Name of the node.
+      op: `OpType` representing the operation type of the node.
+      params: `Params` used by the operator (optional).
+      inputs: A list of `Tensor` (optional).
+      outputs: A list of `Tensor` (optional).
+
+    Returns:
+      A `Node` instance.
+    """
+    self._name = name
+    self._op = op
+    self._params = params
+    self._inputs = []
```

Comment:
Why wouldn't this work?

```python
self._inputs = [] if inputs is None else inputs
```

Are you concerned about assigning `_inputs` directly to `inputs` due to object reference reasons? If so you can just make a copy (`copy.copy(list)`).

Reply:
Here I do want to store the references. So, modified as you suggested.